src/rsa.rs

```diff
@@ -475,6 +486,51 @@ extern {
                                  rng: *mut rand::RAND) -> c::int;
 }
 
+#[cfg(feature = "rsa_signing")]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn BN_rand_range(dest: *mut BIGNUM, range: *const BIGNUM,
+                                   rng: *mut rand::RAND) -> c::int {
```

Comment:
What do you think about this other strategy, which might be simpler.

1. In the C code, create the BIGNUM.
2. Use `bn_wexpand` to expand the BIGNUM's storage to the necessary number of words, `num_limbs`, which comes from the `top` of the modulus.
3. Call a Rust function `GFp_rand_mod(dest: *mut Limb, mod: *const Limb, num_limbs: c::size_t) -> c::int`, which returns zero on failure and non-zero on success.
4. Create a function that uses either `u64_as_u8_mut` or `u32_as_u8_mut` to view `dest` as an array of bytes and use SecureRandom.fill() to fill `dest` with random bytes. (Or, add functions to `SecureRandom for filling arrays of `u32` and `u64`.)
5. Use `GFP_constant_time_limbs_lt_limbs` to ensure that `dest` is less than `m`; Repeat until that is true.

Then, in a separate commit (maybe even a separate PR):

6. Change the ECC code to work a similar way as above, and then have it serialize the resulting `[Limb]` as big-endian bytes. (Instead of generating random bytes, then deserializing the bytes to `[Limb]`, then comparing.)

I think this alternate strategy might greatly reduce the amount of new code you need *for this PR*. And, also, it would avoid the need to worry about allocating very buffers on the stack. Then we'd have more time, at least, to work together on a strategy for dealing with the memory allocation.

I suspect that some and maybe even most of the code you wrote for this PR would then potentially be usable for some future refactoring of the RSA code.

WDYT?